contracts/Vault.sol

```diff
@@ -215,23 +226,28 @@ contract Vault is ExchangeRate, OwnableUpgradeSafe, PausableUpgradeSafe {
     }
 
     /// @notice Convenience function to execute a state transition.
+    /// @dev currently contract transition without manualTransition with the assumption that there are enough deposits or withdraws for constant state changes.
+    /// @dev manual transition could be called any time if enough interests are earned, or if it's time to move onto the next checkpoint.
     function manualTransition() public {
```

Comment:
just noticed that manualTransition is public and has no access roles. doesn't this mean anyone could call it? I guess that'd be fine, but it also seems like an attack vector. do we want to add admin_role to only  be able to do this?

Reply:
manual transition can be callable by anyone, there is no attack vector as far as I know, a caller would simply exhaust gas while helping keeping our balance up to date ✨